experimental/runtimes/src/thread_manager.rs

```diff
@@ -65,3 +66,10 @@ impl ThreadManagerBuilder {
         }
     }
 }
+
+/// This assumes that we get a peak performance at 48 parallelism - in future if we can scale up
+/// our code for higher parallelism, we can increase this number.
+static OPTIMAL_MAX_PARALLELISM: usize = 48;
```

Comment:
Do we want each worker on average have a few tasks rather than one (in the case of our 60 cores experiments)? 
I can see bad cases where we increase concurrent_threads from 48 to 54 and do not see improvements, or one single task being blocked by something else in the system and we have to wait for it.

Reply:
Good point, I actually reduced the max number of threads to 32 and increased this to 64, so on an average, there will be 2 tasks per thread. This reduces the task stealing overhead as well as leaves some room for load balancing in the worst case. This configuration performs optimal in my experiment and gives us around 15% performance boost.

Reply:
2 tasks per thread doesn't sound right, unless the task is super cheap. My intuition is to make one task at 100us to 1ms level.

Reply:
> 2 tasks per thread doesn't sound right, unless the task is super cheap.

Just to be clear, task here is stealable task - and each stealable task might contain ~1000s of small tasks. 

> in other word it should be based on the computation cost of the task, not a global number

Why does it need to be based on the computation cost? We know the parallelism in the system and as long as we know the tasks are uniform, the best strategy is to evenly distribute tasks across threads, leaving some room for task stealing but not allow task stealing on a huge scale that is happening today.

Reply:
You don't want a stealable task to be too large so that one thread is doing it while other threads are waiting and cannot steal.
On the other hand you don't want it to be too small so that the stealing benefit is less than the overhead.

Reply:
Made this 4 * number of threads - trying to find a balance between both of them is tricky for sure.